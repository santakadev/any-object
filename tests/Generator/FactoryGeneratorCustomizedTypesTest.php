<?php

declare(strict_types=1);

namespace Santakadev\AnyObject\Tests\Generator;

use ApprovalTests\Approvals;
use Santakadev\AnyObject\Tests\Generator\Generated\AnyNumberBetweenCustomizedObject;
use Santakadev\AnyObject\Tests\Generator\Generated\AnyRandomDigitCustomizedObject;
use Santakadev\AnyObject\Tests\TestData\CustomizedTypes\NumberBetweenCustomizedObject;
use Santakadev\AnyObject\Tests\TestData\CustomizedTypes\RandomDigitCustomizedObject;

class FactoryGeneratorCustomizedTypesTest extends FactoryGeneratorTestCase
{
    public function test_customized_int_with_number_between(): void
    {
        $this->generateFactoryFor(NumberBetweenCustomizedObject::class);

        $text = $this->readGeneratedAnyFileFor(NumberBetweenCustomizedObject::class);
        Approvals::verifyString($text);
        $this->assertAll(
            fn () => AnyNumberBetweenCustomizedObject::build()->value,
            [
                fn ($value) => $value === 5,
                fn ($value) => $value === 6,
                fn ($value) => $value === 7,
            ]
        );
    }

    public function test_customized_int_with_random_digit(): void
    {
        $this->generateFactoryFor(RandomDigitCustomizedObject::class);

        $text = $this->readGeneratedAnyFileFor(RandomDigitCustomizedObject::class);
        Approvals::verifyString($text);
        $this->assertAll(
            fn () => AnyRandomDigitCustomizedObject::build()->value,
            [
                fn ($value) => $value === 0,
                fn ($value) => $value === 1,
                fn ($value) => $value === 2,
                fn ($value) => $value === 3,
                fn ($value) => $value === 4,
                fn ($value) => $value === 5,
                fn ($value) => $value === 6,
                fn ($value) => $value === 7,
                fn ($value) => $value === 8,
                fn ($value) => $value === 9,
            ]
        );
    }
}
