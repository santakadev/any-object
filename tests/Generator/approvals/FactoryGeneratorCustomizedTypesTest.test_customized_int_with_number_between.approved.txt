<?php

namespace Santakadev\AnyObject\Tests\Generator\Generated;

use Faker\Factory;
use Santakadev\AnyObject\Tests\TestData\CustomizedTypes\NumberBetweenCustomizedObject;

final class AnyNumberBetweenCustomizedObject
{
    public static function with(int|None $value = new None(), int|None $value2 = new None()) : NumberBetweenCustomizedObject
    {
        if ($value instanceof None) {
            $faker = Factory::create();
            $value = mt_rand(5, 7);
        }
        if ($value2 instanceof None) {
            $faker = Factory::create();
            $value2 = mt_rand(PHP_INT_MIN, PHP_INT_MAX);
        }
        return new NumberBetweenCustomizedObject($value, $value2);
    }
    public static function build() : NumberBetweenCustomizedObject
    {
        return self::with();
    }
}
